app/federation/server/functions/helpers.js

```diff
@@ -17,7 +17,7 @@ export function updateEnabled(enabled) {
 	Settings.updateValueById('FEDERATION_Enabled', enabled);
 }
 
-export const checkRoomType = (room) => room.t === 'p' || room.t === 'd';
+export const checkRoomType = (room) => room.t === 'p' || room.t === 'd' || (room.t === 'c' && Settings.findOneById('FEDERATION_Public_Channels'));
```

Comment:
This should be changed to:

```settings.get('FEDERATION_Public_Channels')```

importing the `settings` using:

```import { settings } from 'server/lib/app/settings';```

Please change the path accordingly!

Reply:
Will do!